worker/src/core/loaders/message-worker/email.class.ts

```diff
@@ -21,11 +21,24 @@ class Email {
     )
   }
 
-  enqueueMessages(jobId: number): Promise<void> {
+  enqueueMessages(jobId: number, campaignId: number): Promise<void> {
     return this.connection
-      .query('SELECT enqueue_messages_email(:job_id); ', {
-        replacements: { job_id: jobId },
-        type: QueryTypes.SELECT,
+      .transaction(async (transaction: Transaction) => {
+        await this.connection.query('SELECT enqueue_messages_email(:job_id);', {
+          replacements: { job_id: jobId },
+          type: QueryTypes.SELECT,
+          transaction,
+        })
+        // This is to ensure that stats count tally with total count during sending
+        // as enqueue step may set messages as invalid
+        await this.connection.query(
```

Comment:
why dont you call this in enqueue_messages_email?

Reply:
enqueue_messages_email retrieves the campaign_id

Reply:
But wait, you dont actually need to update_stats here, do you?
```
      error: opsStats.error,
      unsent: opsStats.unsent,
      sent: opsStats.sent + archivedStats.sent,
      invalid: opsStats.invalid + archivedStats.invalid,
      status: job.status,
```
getCurrentStats already accounts for invalid recipients in ops.

Reply:
[02:46] pboy: assume that u have unsent: 100 the rest 0
[02:46] pboy: then enqueue gets run
[02:46] pboy: 5 were marked as invalid
[02:46] pboy:
assume that u have unsent: 100 the rest 0
@pboy#5434 btw this is in statistic table
[02:46] pboy: 95 gets actually enqueued
[02:46] pboy: now u poll for stats
[02:46] jean t: you are preventing overcounting
[02:46] jean t: i see
[02:47] pboy: undercounting
[02:47] pboy: in the ops u have a total of 95 sent + sending + error messages
[02:47] pboy: but 0 invalid
[02:47] pboy: cos the statistic table still has 0 invalid